onnxruntime/core/framework/endian_utils.h

```diff
@@ -28,51 +28,51 @@ namespace utils {
  * @param destination_bytes The destination byte span.
  */
 void SwapByteOrderCopy(
-    size_t element_size_in_bytes, gsl::span<const char> source_bytes, gsl::span<char> destination_bytes);
+    size_t element_size_in_bytes, gsl::span<const uint8_t> source_bytes, gsl::span<uint8_t> destination_bytes);
 
 namespace detail {
 
 /**
  * Copies between two buffers where one is little-endian and the other has
  * native endian-ness.
  */
-void CopyLittleEndian(
-    size_t element_size_in_bytes, gsl::span<const char> source_bytes, gsl::span<char> destination_bytes);
+Status CopyLittleEndian(
+    size_t element_size_in_bytes, gsl::span<const uint8_t> source_bytes, gsl::span<uint8_t> destination_bytes);
 
 }  // namespace detail
 
 /**
  * Reads from a little-endian source.
  */
+common::Status ReadLittleEndian(size_t element_size,
+                                gsl::span<const uint8_t> source_bytes,
+                                gsl::span<uint8_t> destination_bytes);
+
+/**
+ * Reads from a little-endian source with check that T is trivially copyable.
```

Comment:
nit: only true if not GCC below 5

Reply:
if we are stuck with supporting the older GCC versions, maybe we can consider this:
https://stackoverflow.com/a/31798726

Reply:
Not sure that matters. If there's code that is going to fail, it will fail in our CIs which use later versions of GCC won't it, as we don't have any code that is selectively included for a build with GCC < 5?

Reply:
ok, then my original comment applies, regarding "with check that T is trivially copyable."